File: Python3/Roman to Integer.py
# -*- coding: utf-8 -*-
"""
Created on Thu Aug 16 13:55:35 2018

@author: brandt84828
"""

s = "IV"
roman = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
total = 0


for i in range(0,len(s)-1):
    if s[i] =='I' and (s[i+1]=='V' or s[i+1]=='X'):
        total = total - 1
    elif s[i] =='X' and (s[i+1]=='L' or s[i+1]=='C'):
        total = total - 10
    elif s[i] =='C' and (s[i+1]=='D' or s[i+1]=='M'):
        total = total - 100
    else:
        total = total + roman[s[i]]

total = total + roman[s[-1]]


print(total)

File: Golang/RomanToInteger.go
func romanToInt(s string) int {

    roman := map[byte]int{'M':1000, 'D':500, 'C':100, 'L':50, 'X':10, 'V':5, 'I':1}
    total := 0

    for i:=0; i < len(s) - 1; i++ {
        if s[i] == 'I' && (s[i+1] == 'V' || s[i+1] == 'X') {
            total = total - 1
        } else if s[i] == 'X' && (s[i+1] == 'L' || s[i+1] == 'C') {
            total = total - 10
        } else if s[i] == 'C' && (s[i+1] == 'D' || s[i+1] == 'M') {
            total = total - 100
        } else {
            total = total + roman[s[i]]
        }
    }

    total = total + roman[s[len(s) - 1]]

    return total
    
}